README.md

```diff
@@ -1,26 +1,324 @@
+
+## Part II: Controllers and Views
+
+#### Step 1: The Root Path and the Homepage
+
+First things first, let's set up our root path and our home page. 
+
+* Open up `app/controllers/application_controller.rb` and checkout the `get '/'` route. This route should render the `app/views/home.erb` page with the following code: 
+
+```ruby
+erb :'/home'
+```
+* Run your test suite again with `learn` or `rspec` in the command line and you should be passing these two tests: 
+
+```bash
+ApplicationController
+  home page: GET /
+    responds with a 200 status code
+    renders the home page view, 'home.erb'
+```
+* Start up your app by running `shotgun` in the terminal. Visit the home page at `localhost:9393` and you should see message that welcomes you to Hogwarts and shows you a link to sign up and a link to log in.
+
+* Let's look at the code behind this view. Open up that `app/views/home.rb` view and you should see the following
+
+```ruby
+<h1>Welcome to Hogwarts</h1>
+  <h4>Please sign up or log in to access your @hogwarts.edu email account</h4>
+  <a href="/registrations/signup">Sign Up</a>
+  <a href="/sessions/login">Log In</a>
+```
+
+Notice that we have two links, the "Sign Up" link and the "Log In" link. Let's take a closer look: 
+
+* The "href" or destination of the first link is `/registrations/signup`. This means that think link points to the route `get 'registrations/signup'`, which can be found in the Registrations Controller, in `app/registrations_controller.rb`. 
+* The "href" or destination of the second link is `/sessions/login`. This means that think link points to the route `get 'sessions/loging'`, which can be found in the Sessions Controller, in `app/sessions_controller.rb`. 
+
+Let's move on to step 2, the building our user sign up flow. 
+
+#### Step 2: User Sign Up with the Registrations Controller and Views
+
+* Open up `app/registrations_controller.rb` and you should see two route. Let's take a look at the first route, the `get '/registrations/signup'`, which is responsible for rendering the signup template. 
+
+```ruby
+get '/registrations/signup' do
+
+    erb :'/registrations/signup'
+end
+```
+
+* Navigate to `localhost:9383/registrations/signup`. You should see a page that says `"sign up below:"`. Let's make a sign up form! 
+
+* Open up `app/views/registrations/signup.erb`. Our signup form needs a field for name, email and password. It needs to `POST` data to the `'/registrations'` path, so your form action should be `/registrations'` and your form method should be `POST`. 
+
+* Once you've written your form, go ahead and place a `binding.pry` inside the `post '/registrations` route in `app/controllers/registrations_controller.rb`. Then, fill out the form in your browser and hit the `"Sign Up"` button. 
```

Comment:
Can we use `puts params` instead? when using pry in Nitrous with Sinatra things often crash.